src/auth/token-generator.ts

```diff
@@ -242,8 +242,10 @@ export class FirebaseTokenGenerator {
    * @param tenantId The tenant ID to use for the generated Firebase Auth
    *     Custom token. If absent, then no tenant ID claim will be set in the
    *     resulting JWT.
+   * @param useEmulator When true we should generate tokens on behalf of the
+   *     Auth Emulator, which means no signature.
    */
-  constructor(signer: CryptoSigner, public readonly tenantId?: string) {
+  constructor(signer: CryptoSigner, public readonly tenantId?: string, public readonly useEmulator?: boolean) {
```

Comment:
Can we handle all the emulator-related logic in a separate implementation of `CryptoSigner`?

```
interface CryptoSigner {
  algorithm: jwt.Algorithm;
  getAccountId(): Promise<string>;
  sign(buffer: Buffer): Promise<Buffer>;
}

class EmulatedSigner implements CryptoSigner {
  public readonly algorithm = 'none';

  // and so on
}
```

Reply:
Great idea, thanks.